core/src/test/java/nu/yona/server/device/service/DeviceServiceTest.java

```diff
@@ -515,6 +532,78 @@ public void getDeviceAnonymizedId_byId_tryGetNonExistingId_exception()
 		service.getDeviceAnonymizedId(createRichardUserDto(), notAddedDevice.getId());
 	}
 
+	@Test
+	public void removeDuplicateDefaultDevices_singleDevice_noChange()
+	{
+		// Add devices
+		String deviceName1 = "First";
+		OperatingSystem operatingSystem1 = OperatingSystem.ANDROID;
+		UserDevice device1 = addDeviceToRichard(0, deviceName1, operatingSystem1, "Unknown");
+
+		// Verify two devices are present
+		Set<UserDevice> devices = richard.getDevices();
+		assertThat(devices.stream().map(UserDevice::getName).collect(Collectors.toSet()), containsInAnyOrder(deviceName1));
+
+		service.removeDuplicateDefaultDevices(createRichardUserDto(), device1.getId());
+
+		// Assert success
+		assertThat(devices.stream().map(UserDevice::getName).collect(Collectors.toSet()), containsInAnyOrder(deviceName1));
+	}
+
+	@Test
+	public void removeDuplicateDefaultDevices_twoDevicesUseFirst_duplicateRemovedActivitiesMoved()
+	{
+		removeDuplicateDefaultDevicesFirstOrSecond(0);
+	}
+
+	@Test
+	public void removeDuplicateDefaultDevices_twoDevicesUseSecond_duplicateRemovedActivitiesMoved()
+	{
+		removeDuplicateDefaultDevicesFirstOrSecond(1);
+	}
+
+	private void removeDuplicateDefaultDevicesFirstOrSecond(int indexToRetain)
```

Comment:
I'd rather use `@Parameters` here.

Reply:
That can't be combined easily with ``SpringJUnit4ClassRunner``. There are tricks for it, but that won't make the test more readable.